addons/mail_group/static/src/js/mail_group.js

```diff
@@ -0,0 +1,83 @@
+odoo.define('mail_group.mail_group', function (require) {
+'use strict';
+
+const publicWidget = require('web.public.widget');
+const core = require('web.core');
+const _t = core._t;
+
+publicWidget.registry.MailGroup = publicWidget.Widget.extend({
+    selector: '.o_mail_group',
+    events: {
+        'click .o_mg_subscribe_btn': '_onSubscribeBtnClick',
+    },
+
+    /**
+     * @override
+     */
+    start: function () {
+        this.mailgroupId = this.$target.data('id');
+        this.isMember = this.$target.data('isMember') || false;
+        return this._super.apply(this, arguments);
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     */
+    _onSubscribeBtnClick: async function (ev) {
+        ev.preventDefault();
+        const $email = this.$target.find(".o_mg_subscribe_email");
+        const email = $email.val();
+
+        if (!email.match(/.+@.+/)) {
+            this.$target.addClass('o_has_error').find('.form-control, .custom-select').addClass('is-invalid');
+            return false;
+        }
+
+        this.$target.removeClass('o_has_error').find('.form-control, .custom-select').removeClass('is-invalid');
+
+        let action = this.isMember ? 'unsubscribe' : 'subscribe';
+
+        const response = await this._rpc({
+            route: '/groups/subscription',
+            params: {
+                'group_id': this.mailgroupId,
+                'action': action,
+                'email': email,
+            },
+        });
+
+        this.$el.find('.o_mg_alert').remove();
+
+        if (response === 'done') {
+            this.isMember = !this.isMember;
```

Comment:
Would return explicitly "added" or "removed". We have issues with people spamming "follow" button on frontend, using a key allows to avoid a de-synchronization when people click several times on a button. Indeed inner status (isMember) could toggle several times although actually it should only be False or True depending on controller result.

Reply:
Good idea :)